docs/mshv.md

```diff
@@ -0,0 +1,52 @@
+
+Microsoft till now has only run Windows as the root OS. Companies like Amazon, Google and Alibaba have long been using Linux as the primary driver for their cloud offerings. Linux is pervasive on both the host and guest side in their clouds. These companies and other open source developers have made many contributions to the Linux OS for supporting virtualization technology and Linux has become feature rich in this area. By limiting the root OS to Windows, Microsoft is missing out on these innovations that can be leveraged to support a vast array of Microsoft’s business needs, with quick turnaround solutions.
+
+In the recent years, Microsoft has added Linux as Dom-0 support to the Microsoft hypervisor. It open sourced different components as follows
+
+    1. Added support to linux tree to boot on Microsoft Hyper-V
+    2. Linux module to mane VMs (mshv)  that exposes /dev/mshv
+    3. Rust equivalent bindings and IOCTL wrappers
+    4. IGVM related crates
+
+## Components
+
+The following components are related to MSHV support with Cloud-Hypervisor:
+
+* [igvm-crates](https://github.com/microsoft/igvm) : Parsing IGVM file
+
+* [mshv-crates](https://github.com/rust-vmm/mshv) : Rust crates to interact with Kernel Module (/dev/mshv)
+
+* [igvm-tooling](https://github.com/microsoft/igvm-tooling) : Tool to generate IGVM file
+
+## IGVM
+
+Independent Guest Virtual Machine (IGVM) file format.The format specification can be found in the igvm_defs crate, with a Rust implementation of the binary format in the igvm crate.
+
+The IGVM file format is designed to encapsulate all information required to launch a virtual machine on any given virtualization stack, with support for different isolation technologies such as AMD SEV-SNP and Intel TDX.
+
+At a conceptual level, this file format is a set of commands created by the tool that generated the file, used by the loader to construct the initial guest state. The file format also contains measurement information that the underlying platform will use to confirm that the file was loaded correctly and signed by the appropriate authorities.
+
+Cloud Hypervisor can be built using igvm feature flag along with mshv and/or sev-snp
+
+## SEV-SNP
+
+ AMD introduced Secure Encrypted Virtualization (SEV). It also added Platform Security Processor (PSP) firmware
+support for the Secure Nested Paging (SEV-SNP) enhancement to SEV. Microsoft has been continuously adding/improving
+support for SEV-SNP on Microsoft Hyper-V. Cloud-Hypervisor can be built with this feature including mshv and igvm feature.
+
+## Feature dependency
+
+Cloud-Hypervisor on MSHV can boot a simple linux guest with most of the devices capability like kvm.
```

Comment:
Not sure what is meant here exactly... just that it does not depend on any other features?
"Cloud Hypervisor can be built to run on an MSHV root partition by enabling the `mshv` feature, e.g.:"